lib/sms_signup.rb

```diff
@@ -0,0 +1,32 @@
+class SmsSignup
+  def initialize(user_model:)
+    @user_model = user_model
+  end
+
+  def execute(contact:)
+    phone_number = normalised_phone_number(contact)
+    login_details = user_model.generate(contact: phone_number)
```

Comment:
generate to me sounds like something stored in memory. find_or_create_by feels like a better name to me.  Probably a refactoring on its own

Reply:
Yeah, that was my view.  See this comment here: https://github.com/alphagov/govwifi-user-signup-api/pull/7#discussion_r197150844